volatility3/framework/plugins/linux/ip.py

```diff
@@ -47,15 +47,15 @@ def _gather_net_dev_info(self, net_dev: net_extensions.net_device):
             prefix_len = in_ifaddr.get_prefix_len()
             scope_type = in_ifaddr.get_scope_type()
             ip_addr = in_ifaddr.get_address()
-            yield net_ns_id, iface_ifindex, iface_name, mac_addr, promisc, ip_addr, prefix_len, scope_type, operational_state
+            yield net_ns_id or renderers.NotAvailableValue(), iface_ifindex, iface_name, mac_addr, promisc, ip_addr, prefix_len, scope_type, operational_state
 
         # Interface IPv6 Addresses
         inet6_dev = net_dev.ip6_ptr.dereference().cast("inet6_dev")
         for inet6_ifaddr in inet6_dev.get_addresses():
             prefix_len = inet6_ifaddr.get_prefix_len()
             scope_type = inet6_ifaddr.get_scope_type()
             ip6_addr = inet6_ifaddr.get_address()
-            yield net_ns_id, iface_ifindex, iface_name, mac_addr, promisc, ip6_addr, prefix_len, scope_type, operational_state
+            yield net_ns_id or renderers.NotAvailableValue(), iface_ifindex, iface_name, mac_addr, promisc, ip6_addr, prefix_len, scope_type, operational_state
```

Comment:
I thought we were favouring returning `None` and then at render time doing `or NotAvailableValue()`?

Reply:
I left this as acceptable here b/c the function is just gathering the data directly sent through the generator. If we make it return None in the line you highlighted then this line in the generator (this is a link to the full file) would need to unpack all those fields (10 or so) versus just passing it through:

https://github.com/volatilityfoundation/volatility3/blob/6a9a7bc1c6b0b1c5b90c03aec5b71ab7ab39dd23/volatility3/framework/plugins/linux/ip.py#L71

Reply:
Mmmmm, not sure how I feel about that.  It's a good reason but given this is going to be reused a *lot*, I'd prefer we did the unpacking (once) rather than letting the RendererValues sneak further into the API.  It's much harder to convince people not to do this stuff, if they can point to a core example doing it...  5:S  It's a little more work now, but it'll keep everything future and dependable in the future.  Could you make the changes please?